.github/workflows/playwright.yml

```diff
@@ -0,0 +1,42 @@
+name: Playwright Test
+on: push
+
+jobs:
+  e2e-tests:
+    name: Playwright test
+    runs-on: ubuntu-latest # or macos-latest, windows-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 'latest'
+      # Use the official setup-node action (sets up Node.js):
+      # https://github.com/actions/setup-node
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 'lts/*'
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install
+      - name: Build Next.js app
+        run: pnpm build
+      - name: Get installed Playwright version
+        id: playwright-version
+        run: echo "PLAYWRIGHT_VERSION=$(yq eval '.devDependencies["@playwright/test"]' --output-format=yaml ./package.json)" >> $GITHUB_ENV
+      - name: Cache playwright binaries
+        uses: actions/cache@v3
+        id: playwright-cache
+        with:
+          path: |
+            ~/.cache/ms-playwright
+          key: ${{ runner.os }}-playwright-${{ env.PLAYWRIGHT_VERSION }}
```

Comment:
Sets the PLAYWRIGHT_VERSION variable by getting the version of `@playwright/test` from `package.json` using `yq`.
Caches the Playwright directory `~/.cache/ms-playwright` based on the operating system `(${{ runner.os }})` and the `PLAYWRIGHT_VERSION` environment variable.